libr/util/strbuf.c

```diff
@@ -119,14 +121,32 @@ R_API bool r_strbuf_slice(RStrBuf *sb, int from, int len) {
 	return true;
 }
 
-R_API bool r_strbuf_set(RStrBuf *sb, const char *s) {
+R_API bool r_strbuf_setptr(RStrBuf *sb, char *s, int len) {
```

Comment:
```suggestion
R_API bool r_strbuf_setptr(RStrBuf *sb, char *s, ssize_t len) {
```

Reply:
why?

Reply:
Why not?

Reply:
And why not even to use `size_t`?

Reply:
what about having r_strbuf_setptr(const ut8* , size_t len); and r_strbuf_setstr(const char *str); instead of abusing the length with negative values